test/Microsoft.AspNetCore.SignalR.Tests/HubConnectionHandlerTests.cs

```diff
@@ -2019,6 +2019,73 @@ public int Serialize(ref byte[] bytes, int offset, T value, IFormatterResolver f
             }
         }
 
+        [Fact]
+        public async Task ConnectionNotTimedOutIfClientNeverPings()
+        {
+            var serviceProvider = HubConnectionHandlerTestUtils.CreateServiceProvider(services =>
+                services.Configure<HubOptions>(options =>
+                   options.ClientTimeoutInterval = TimeSpan.FromMilliseconds(100)));
+            var connectionHandler = serviceProvider.GetService<HubConnectionHandler<MethodHub>>();
+
+            using (var client = new TestClient(new JsonHubProtocol()))
+            {
+                var connectionHandlerTask = await client.ConnectAsync(connectionHandler);
+                await client.Connected.OrTimeout();
+                // This is a fake client -- it doesn't auto-ping to signal
+
+                await Task.Delay(120);
+                client.TickHeartbeat();
+
+                await Assert.ThrowsAsync<TimeoutException>(() => connectionHandlerTask.OrTimeout(60));
```

Comment:
What is being tested by this timeout? Could you add a comment

Reply:
checks that the client remains connected (ie that the handler task times out)
only clients that ping (new version) should be timed out